### PR TITLE
Feature/2.x/mongodb next gg

### PR DESCRIPTION
**Pull request by garydgregory**
Mongo DB PR with 3 commits
- Add a new `log4j-mongodb` module to track the current MongoDB Driver, version 5
- Deprecate `log4j-mongodb4` in favor of `log4j-mongodb`
- Remove `log4j-mongodb3`

(it's fine w me to squash on merge)

### CODE REVIEW

File: log4j-mongodb/src/main/java/org/apache/logging/log4j/mongodb/package-info.java [hunk 1]
```diff
@@ -15,12 +15,13 @@
...
 /**
- * The classes in this package contain the MongoDB provider for the NoSQL Appender.
+ * The classes in this package contain the MongoDB provider for the NoSQL
+ * Appender.
  */
 @Export
 @Open("org.apache.logging.log4j.core")
-@Version("2.20.1")
-package org.apache.logging.log4j.mongodb3;
+@Version("2.21.0")
```
Comment (ppkarwasz) [inline log4j-mongodb/src/main/java/org/apache/logging/log4j/mongodb/package-info.java:23]: Let us use `2.24.0`, i.e. the version when the class was introduced. It is then easier to find the last Log4j version with some real changes to the module.

**Review by ppkarwasz — APPROVED** [review]
LGTM

File: log4j-mongodb/src/main/java/org/apache/logging/log4j/mongodb/package-info.java (hunk 1 above)
Comment (garydgregory) [reply log4j-mongodb/src/main/java/org/apache/logging/log4j/mongodb/package-info.java:23]: Good catch!

File: pom.xml [hunk 2]
```diff
@@ -257,8 +257,8 @@
     <module>log4j-jul</module>
     <module>log4j-layout-template-json</module>
     <module>log4j-layout-template-json-test</module>
-    <module>log4j-mongodb3</module>
     <module>log4j-mongodb4</module>
+    <module>log4j-mongodb</module>
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply pom.xml:261]: Maven should reorder.

File: log4j-mongodb/src/test/java/org/apache/logging/log4j/mongodb/MongoDbResolver.java [hunk 3]
```diff
@@ -142,11 +144,11 @@ private static final class MongoClientHolder implements CloseableResource, Suppl
         private final MongoClient mongoClient;
 
         public MongoClientHolder(final Mongod mongod, final TestProperties props) {
-            state = mongod.start(Version.Main.V3_6);
+            state = mongod.start(Version.Main.V4_4);
```
[the remark this answers is not in the document]
Comment (garydgregory) [reply log4j-mongodb/src/test/java/org/apache/logging/log4j/mongodb/MongoDbResolver.java]: That did not even compile. That's what I get for using GH! ;-)

**Pull request merged**